packages/nimble-components/src/anchor-menu-item/index.ts

```diff
@@ -0,0 +1,110 @@
+import { attr, observable } from '@microsoft/fast-element';
+import {
+    DesignSystem,
+    AnchorOptions,
+    MenuItem as FoundationMenuItem
+} from '@microsoft/fast-foundation';
+import { keyEnter } from '@microsoft/fast-web-utilities';
+import { AnchorBase } from '../anchor-base';
+import { styles } from './styles';
+import { template } from './template';
+
+declare global {
+    interface HTMLElementTagNameMap {
+        'nimble-anchor-menu-item': AnchorMenuItem;
+    }
+}
+
+/**
+ * Types of anchor menu item column count.
+ * This is how many grid columns precede the menu item text.
+ * 1 corresponds to having space for an icon, 0 corresponds to the text being flush left.
+ * Users should not need to use this type. The menu sets startColumnCount on its items.
+ * @public
+ */
+export type AnchorMenuItemColumnCount = 0 | 1;
+
+/**
+ * A nimble-styled anchor menu-item
+ */
+export class AnchorMenuItem extends AnchorBase {
+    @attr({ mode: 'boolean' })
+    public disabled = false;
+
+    @observable
+    public anchor!: HTMLAnchorElement;
+
+    @observable
+    public startColumnCount: AnchorMenuItemColumnCount = 0;
+
+    // The following two handlers are workarounds for issues with anchor menu items in submenus.
+    // Events can bubble up the DOM and get handled by the menu item in the parent menu. When that happens,
+    // the menu item's handlers (FAST) return false and prevent the default action, i.e. navigation.
+
+    public clickHandler = (e: MouseEvent): boolean => {
+        e.stopImmediatePropagation();
+        return true;
+    };
+
+    public keydownHandler = (e: KeyboardEvent): boolean => {
+        if (e.defaultPrevented) {
+            return false;
+        }
+        switch (e.key) {
+            case keyEnter:
+                e.stopImmediatePropagation();
+                break;
+            default:
+        }
+        return true;
+    };
+
+    // The FAST Menu manages the `tabindex` of its menu items. Because we're setting `delegatesFocus=true`
+    // the focusable element is the anchor, so we forward `tabindex` to that element instead of the host.
+    // This prevents a problem where the focused anchor loses focus and the menu closes.
```

Comment:
It sounds at least related to [this bug I filed to chromium](https://bugs.chromium.org/p/chromium/issues/detail?id=1346606). Is this also a chromium-only issue as well? They've fixed the bug, but it hasn't made it into a released version of Chrome or Edge yet.

Reply:
The bug molly linked has a stackblitz. I was still seeing that issue in stackblitz with Chrome 110. I updates to the latest shipping Chrome 111 (released Mar 1) by checking for updates in the about dialog and the stackblitz no longer shows the issue.

Reply:
Ah, yes, it is working in the latest Chrome. My version of Edge (110.0.1587.69) claims to be up-to-date, but it doesn't include the fix yet.

Reply:
Yeah, that was exactly the problem I was running into. I confirmed that it doesn't happen in Firefox, and it stopped happening when I updated my version of Chrome. Based on that, I'm going to remove the workaround.

Reply:
We support Chrome, Firefox, Edge, and Safari so does that mean we will have a know issue that this fails to work in Edge?

I confirmed that Molly's stackblitz for the original bug is still an issue in Edge and my Edge reports is up to date (as Molly repro'd as-well).

If we are making it a known issue then it should be a known issue, i.e. create a bug in nimble documenting the bug in Edge and state that it will be resolved in the next version of Edge. I'd feel more confident about that if it was already seen as fixed in an Edge beta, etc. This is probably okay because, at least according to spec, this feature of anchor-menus isn't actually used in product anywhere in SystemLink (is it true we will not use this anywhere after merging?).

The alternative is that we include the workaround and remove it when addressed in latest Edge. Either approach seems reasonable, I'd like @jattasNI's opinion on that.

Reply:
Unmarked resolved

Reply:
I've restored the workaround and created a tech debt item to remove the workaround once the fix makes it into Edge as well. (#1118)